### PR TITLE
User Contributions Demo App

### PR DESCRIPTION
**Pull request by Jayprakash-SE**
A sample app that uses MediaWiki Action API:Usercontribs allows you to see the latest top 50 edits made by a user. This app uses Flask framework as backend and Bootstrap as frontend.

### CODE REVIEW

**Comment by srish** [comment]
@Jayprakash-SE Cool! But before working on the demo apps, please discuss the idea with me first. The task I shared around ideas was just for reference. Some of the ideas in here are already a work in progress https://phabricator.wikimedia.org/T212006 :) Thanks!

**Comment by Jayprakash-SE** [comment]
> @Jayprakash-SE Cool! Before working on the demo apps, please discuss the idea with me first. The task I shared around ideas was just for reference. Some of the ideas in here are already a work in progress https://phabricator.wikimedia.org/T212006 :)
> 
> I will come back to review this in a few days.

Srishti Ji, You can merge this branch, I will create another proposal for GSoC.

File: python/demos/UserContributions/templates/user_contributions.html [hunk 1]
```diff
@@ -0,0 +1,58 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="utf-8" content="width=device-width, initial-scale=1.0" name="viewport">
+
+    <!-- Page title -->
+    <title>User Contributions - Demo app</title>
+
+    <!-- Bootstrap CSS -->
+    <link href="https://tools-static.wmflabs.org/cdnjs/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css"
```
Comment (srish) [inline python/demos/UserContributions/templates/user_contributions.html]: For the sake of demo app, might not be a good idea to link to an external css file that lives on Toolforge..

File: python/demos/UserContributions/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# User Contributions
+A sample app that uses MediaWiki Action [API:Usercontribs](https://www.mediawiki.org/wiki/API:Usercontribs) allows you to see the latest top 50 edits made by a user. This app uses Flask Framework as backend and Bootstrap as frontend.
```
Comment (srish) [inline python/demos/UserContributions/README.md]: May be you don't need to mention Bootstrap here, as you are not really using the framework.

File: python/demos/UserContributions/app.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    username = request.args.get('username')
+
+    # Check whether the username present in request or not
+    if username is not None:
+        data = get_user_contribs(username)
+    else:
+        data = None
+
+    # render the template
+    return render_template('user_contributions.html', data=data, username=username, wikiurl=WIKI_URL)
```
Comment (srish) [inline python/demos/UserContributions/app.py]: Might we don't need to send username data as we are already have it in the response object.

**Comment by Jayprakash-SE** [comment]
https://github.com/wikimedia/MediaWiki-Action-API-Code-Samples/pull/89 :(

**Comment by srish** [comment]
@Jayprakash-SE I see that you've created another pull request around the same topic. Is there a reason for doing so? It would be helpful if you could reply and resolve code review comments you've already addressed and if you've any follow-up question on other comments :)

**Pull request closed without merging**